src/Components/test/E2ETest/ServerExecutionTests/ServerReconnectionTest.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using BasicTestApp;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures;
+using Microsoft.AspNetCore.Components.E2ETest.ServerExecutionTests;
+using Microsoft.AspNetCore.E2ETesting;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Support.UI;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Components.E2ETest.ServerExecutionTests
+{
+    public class ServerReconnectionTest : BasicTestAppTestBase
+    {
+        public ServerReconnectionTest(
+            BrowserFixture browserFixture,
+            ToggleExecutionModeServerFixture<Program> serverFixture,
+            ITestOutputHelper output)
+            : base(browserFixture, serverFixture.WithServerExecution(), output)
+        {
+        }
+
+        public List<(Extensions.Logging.LogLevel level, string eventIdName)> Messages { get; private set; }
+
+        public string SessionIdentifier { get; set; } = Guid.NewGuid().ToString();
+
+        protected override void InitializeAsyncCore()
+        {
+            Navigate(ServerPathBase, noReload: false);
+
+            Browser.Manage().Cookies.DeleteCookieNamed("WebSockets.Identifier");
+            Browser.Manage().Cookies.AddCookie(new Cookie("WebSockets.Identifier", SessionIdentifier));
+            Browser.Navigate().Refresh();
```

Comment:
Do we want to just replace this with navigation + reload? (line 34)

Reply:
Did you mean line 32? I don't see why

Reply:
I mean line 30. Do we need to navigate to the root **and then** trigger a refresh?

Reply:
In the server-side case yes because the connection already got established by the time we set the cookies.

These tests don't run client-side.